modules/assistant/session.go

```diff
@@ -264,8 +265,23 @@ func (h APIHandler) createChatSession(w http.ResponseWriter, r *http.Request, ps
 	inflightMessages.Store(replyMsgTaskID, MessageTask{
 		SessionID:  session.ID,
 		CancelFunc: cancel,
+		CreatedAt:  time.Now(),
 	})
-	_ = h.processMessageAsync(ctx, reqMsg, params, streamSender)
+
+	// Process message asynchronously and cleanup on completion
+	go func() {
+		defer func() {
+			// Always cleanup the task when processing completes (success or failure)
+			// Load and delete atomically to handle race conditions with external cancellation
+			if taskValue, loaded := inflightMessages.LoadAndDelete(replyMsgTaskID); loaded {
+				if messageTask, ok := taskValue.(MessageTask); ok && messageTask.CancelFunc != nil {
```

Comment:
processMessageAsync is running in background tasks, defer here will cancel the task immediately.